examples/sqlite.rs

```diff
@@ -233,7 +233,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .layer(HandleErrorLayer::new(|_: BoxError| async {
             StatusCode::BAD_REQUEST
         }))
-        .layer(AuthManagerLayer::new(backend, session_layer));
+        .layer(AuthManagerLayer::new(backend, "session", session_layer));
```

Comment:
One consideration here is that this lives in the same session that application author's will access via `tower-sessions` directly. For that reason, it might be good to prefix it with a namespace that's unlikely to clash with arbitrary keys that live in the session alongside it. So I would recommend something like "axum-session.[something]" or similar.

Reply:
I changed it to `axum-session.example` so it's clear that it should be modified.

Additionally maybe we'd like a `AuthManagerLayerBuilder` to be public and the `AuthManagerLayer` to be only buildable with that builder? That would allow us to add any number of additional parameters in the future without causing an API brake. What do you think?

Reply:
> I changed it to axum-session.example so it's clear that it should be modified.

I might be missing an obvious use case here: I've been thinking about this as primarily an internal implementation detail that `AuthSession` encapsulates. Generally speaking, my thought was that application authors wouldn't need to change the key name (with the exception of key collisions, but assuming a good prefix that effectively namespaces the name, I think that should be unlikely). I'm not opposed to providing the flexibility, but I don't think we'd generally want people to change this. Does that track with you?

Reply:
Yep my main concern was collisions, so it should be configurable somehow. In reality I'd go for a safe default with some builder (that why I proposed it) and leave the user with the ability to change it.

Reply:
That seems reasonable to me. Could we use `"axum-login.data"` as a default in that case? (This name maps the name of the struct it stores, `Data`.)

Reply:
Sure, implement it as soon as I have a moment.

Reply:
No rush at all. Thank you for all the contributions here. 🙇

Reply:
That seems good to me. (I also wonder if we should do something similar for `tower-sessions`?)

Reply:
I believe it's a good idea - less possible headaches when adding new optional configurations in the future ;). If someone uses the layer directly, their code may break if we add a new parameter. With a private `new` method and a builder, we can always provide some safe default and all code that worked, will still behave the same way.